test/net_http_test.rb

```diff
@@ -511,42 +492,15 @@ def close_circuit!(hostname: SemianConfig['toxiproxy_upstream_host'], toxic_port
 
   def with_server(ports: [SemianConfig['http_port']], reset_semian_state: true)
     ports.each do |port|
-      hostname = '0.0.0.0'
-      begin
-        server = nil
-        server_threw_error = false
-        server_thread = Thread.new do
-          Thin::Logging.silent = true
-          server = Thin::Server.new(hostname, port, RackServer)
-          begin
-            server.start
-          rescue StandardError
-            server_threw_error = true
-            raise
-          end
-        end
-
-        begin
-          poll_until_ready(hostname: hostname, port: port)
-        rescue RuntimeError
-          server_thread.kill
-          server_thread.join if server_threw_error
-          raise
-        end
-
-        assert(server.running?)
-        reset_semian_resource(hostname: SemianConfig['toxiproxy_upstream_host'], port: port) if reset_semian_state
+      if port == SemianConfig['http_port'] || SemianConfig['http_port'] + 100 # rubocop:disable all
```

Comment:
Similar to another comment: what is this 100? We should avoid magic numbers, putting them into something that is named (constant, function, etc).